runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1217,6 +1217,16 @@ gcParseSovereignArguments(J9JavaVM *vm)
 	if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:-HeapManagementMXBeanCompatibility", NULL)) {
 		extensions->_HeapManagementMXBeanBackCompatibilityEnabled = false;
 	}
+
+	if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:+UseGCStartupHints", NULL)) {
+
+		extensions->useGCStartupHints = true;
+	}
+
+	if (-1 != FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XX:-UseGCStartupHints", NULL)) {
```

Comment:
Typically the order of options matter, and only the last one takes affect.

Reply:
I'm not sure I understand your point. Are you suggesting I should reorder them in source code and why?

Yes, in GC right most one wins because we consume them from left to right. I believe it has nothing to do with how in source code they are ordered.

Reply:
This is correct. An order in the code is irrelevant - control code would do correct thing: most right option wins

Reply:
There are two options being searched here, you need to compare the index of each option and only do the action for the option with the higher index. The way this is written, if the command line contains `-XX:-UseGCStartupHints -XX:+UseGCStartupHints` you would expect startup hints to be enabled, but the startup hints will be disabled because `-XX:-UseGCStartupHints` is checked last.

Reply:
https://github.com/eclipse/openj9/blob/e2f7d273c2317a4d99f00f2fc626df55e62cef4a/runtime/verbose/verbose.c#L688-L693

Reply:
You are right. I surprised this code was not fixed. We do have `while` loop to parse options for `-Xgc:` and `-XXgc:` to handle order, so I thought such loop exists there as well... but it is not correct

Reply:
I fixed it, but I just did minimal changes (did not introduce the while loop, like we have for -Xgc:).

Reply:
No loops are required here, the looping is done in FIND_ARG_IN_VMARGS

Reply:
I understand, it's not required. But for -Xgc: and -XXgc: options we do have an extra loop around FIND  that help us (without going into details) avoid this game with indices. I was just saying (mostly to Dmitri(, that here it's not as easy to apply that logic (because we cannot recognize GC specific options within -XX:  as we can within the two groups above) and simply adopted your pattern instead.